### ISSUE TITLE
W0233 non-parent-init-called triggered by call to `super().__init__()`

### ISSUE THREAD

**Issue by JoeHCQ1** [PyCQA/pylint#5914]
### Bug description

I have a series of classes inheriting from each other, all single inheritance. This code is triggering these errors:

```
src/foo.py:31:8: W0233: __init__ method from a non direct base class 'super' is called (non-parent-init-called)
src/foo.py:28:4: W0231: __init__ method from base class 'Bar' is not called (super-init-not-called)
```

```python
class Foo(Bar):

    def __init__(self):
        self._foo = None
        self._bar = None
        super().__init__()
```

If I'm not mistaken, `super` is how you should call your parent class' constructor rather than hardcoding the value so you don't break python's dynamic call order in the case of multiple inheritance.

### Command used

```shell
python -m pylint --init-hook="sys.path.append('.')" src/ other_src/
```

### Pylint output

```shell
src/foo.py:31:8: W0233: __init__ method from a non direct base class 'super' is called (non-parent-init-called)
src/foo.py:28:4: W0231: __init__ method from base class 'Bar' is not called (super-init-not-called)
```

### Expected behavior

Expect no error.

### Pylint version

```shell
pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04 LTS.

### Additional dependencies

```
# required
cffi
numpy

# testing
mypy
flake8
pytest
pytest-cov
pylint
rope
autopep8
velin==0.0.11

# install
wheel

# Generate documentation
mkdocs>=1.2  # There is a breaking change between 1.1 and 1.2 regarding livereload
pytkdocs[numpy-style]  # Required so mkdocstrings can read numpy-style docstrings
mkdocstrings>=0.18.0

# Change this to `mkdocstrings-python` once that new python module works.
mkdocstrings-python-legacy

mkdocs-git-revision-date-localized-plugin>=0.12
mkdocs-macros-plugin>=0.6.4
mkdocs-material
mkdocs-material-extensions
mkdocs-mermaid-plugin
mkdocs-pdf-export-plugin
mkdocs-autorefs
mkdocs-gen-files>=0.3.3
mkdocs-literate-nav
mkdocs-section-index
```

**Comment by jacobtylerwalls** [comment]
Hi, thanks for the report. I can't reproduce the issue with the provided example.

> If I'm not mistaken, super is how you should call your parent class' constructor rather than hardcoding the value so you don't break python's dynamic call order in the case of multiple inheritance.

Yes, that's true--instead, it appears the ancestor classes were not inferred correctly. We will need more information about your classes to reproduce. Would you be able to provide a minimal reproducer? Thanks!

**Comment by JoeHCQ1** [comment]
I can no longer reproduce it.

**Comment by mattig7** [comment]
For anyone's future information, I got this same error when I called 'super.__init__()' rather than 'super().__init__()'.
I suspect it is the difference between calling the type object of the superclass and an instance of the superclass (hence it is calling __init__ on a class that isn't in the ancestor chain)
